examples/PERFORMANCE.md

```diff
@@ -0,0 +1,181 @@
+
+The DSE can be asked to act smarter than in `advanced` mode by setting it to
+`aggressive`. This can be done by providing the kwarg `dse='aggressive'` to an
+Operator, or globally by setting
+```
+DEVITO_DSE=aggressive
+```
+In the latter case, however, consider that local choices take precedence.
+Hence, if `dse=advanced` is passed to the Operator `OP`, then `OP` will be
+compiled in `advanced` mode, even though `DEVITO_DSE=aggressive` was set.
+One may understand that the switch to aggressive mode was successful by
+observing changes in the Devito DSE output, printed to screen when code
+generation for an Operator takes place.
+
+Significant reductions in operation count due to using `aggressive` mode have
+been observed in several TTI examples. The `aggressive` mode may or may not
+increase the Devito processing time.
+
+### Auto-tuning
+
+Operator auto-tuning can greatly improve the run-time performance. It can be
+activated on an Operator basis by passing the flag `autotune=True` at
+Operator application time:
+```
+op = Operator(...)
+op.apply(autotune=True)
+```
+The auto-tuning system will then attempt to retrieve the optimal block size for
+loop blocking. When seeking maximum performance, it is recommended to maximize the
+auto-tuner aggressiveness through:
+```
+DEVITO_AUTOTUNING=aggressive
+```
+
+### Choice of the backend compiler
+
+For each Operator, Devito generates C code, which then gets compiled into a
+shared object through the user-selected "backend compiler". Supported backend
+compilers are `gcc`, `icc`, `llvm`. In theory, any C compiler would work, but
+for the aforementioned ones, Devito is aware of the best compilation flags. If
+multiple compilers are available on the system, one can make a selection by
+exporting
+```
+DEVITO_ARCH=X
+```
+`print_defaults()`, again, provides a list of legal values for X.
+
+For maximum performance, it is *highly recommended to use the Intel compiler*.
+Devito performs SIMD vectorization by resorting to the backend compiler
+auto-vectorizer, and Intel's is particularly effective in stencil codes.
+
+### Be aware of what's happening in Devito
+
+Run with
+```
+DEVITO_LOGGING=DEBUG
+```
+To get more info from Devito about the performance optimizations applied or
+on how auto-tuning is getting along.
+
+# Known limitations and possible work arounds
+
+ * At the moment, there is no support for MPI parallelism. This is perhaps the
+   biggest limitation, because it significantly affects execution on
+   multi-socket nodes.  The ideal setting on multi-socket nodes would be to
+   have 1 MPI process per socket, and OpenMP (or, why not, MPI itself in
+   shared-memory mode) for the processes within a socket. One can still use
+   OpenMP across all available sockets (the default case if `OMP_NUM_PROCS` is
+   unset and more than one sockets are available), but the final performance
+   will be very far from the attainable machine peak, due to the well known
+   NUMA effect.
```

Comment:
We do have a parameter `numa` that can be set on `DenseData`/`TimeData` objects to force first-touch, but I don't think is enabled by default, or accessible through global parameters. Should we maybe add this and mention it here?